Updated terminology (#2815)
dfinity/motoko: doc/modules/language-guide/pages/motoko.adoc

{empty}

[.card]
== Native canister smart contract support

> Motoko has native support for Internet Computer canister smart contracts, which are expressed as actors, autonomous objects that encapsulate their state and communicate through asynchronous messages.

Reviewer: ```suggestion
Motoko has native support for Internet Computer canister smart contracts, which are expressed as Motoko actors, autonomous objects that encapsulate their state and communicate through asynchronous messages.
```

[.card]
== Code sequentially in direct style

> On the Internet Computer, canister smart contracts call into other canisters smart contracts asynchronously, but Motoko enables you to program your systems sequentially in direct style. Asynchronous messages are function calls that return a future, and the await constructs allows chaining calls as if they were synchronous.

Reviewer 2: ```suggestion
On the Internet Computer, canister smart contracts call into other canister smart contracts asynchronously, but Motoko enables you to program your systems sequentially in direct style. Asynchronous messages are function calls that return a future, and the await constructs allows chaining calls as if they were synchronous.
```

[.card]
== Autogenerated IDL files

> The SDK exports your interface definition in a language neutral format called `+Candid+`, so other canister smart contracts, browser resident code and smart phone apps that have permission can call into your functions. The Motoko compiler can also read and write interface definition files, allowing Motoko to seamlessly interact with canister smart contracts programmed in other languages.

Reviewer: ```suggestion
The SDK exports your interface definition in a language neutral format called `+Candid+`, so other canister smart contracts, browser resident code and smart phone apps that have permission can call into your functions. The Motoko compiler can also read and write interface definition files, allowing Motoko to seamlessly interact with canister smart contracts implemented in other programming languages.
```

[.card]
== Orthogonal persistence

> The Internet Computer persists the memory pages in which your canister smart contract runs. Thus the state of an actor and all its memory data structures survive indefinitely, they do not need to be “saved” explicitly.

Reviewer: ```suggestion
The Internet Computer persists the memory pages and other state of your canister smart contract as it executes. Thus the state of an actor and all its memory data structures survive indefinitely, they do not need to be “saved” explicitly.
```

> For example, in the following Registry canister smart contract, that assigns sequential IDs to textual names, the state of the hash table is preserved across calls, even though the state of the canister smart contract

Reviewer: ```suggestion
For example, in the following Registry actor (canister), that assigns sequential IDs to textual names, the state of the hash table is preserved across calls, even though the state of the actor
```

[.card]
== Upgrades

> Motoko provides numerous features to help you leverage orthogonal persistence, including language features that allow your heap to self-migrate when you upgrade the software of a canister smart contract.

Reviewer: ```suggestion
Motoko provides numerous features to help you leverage orthogonal persistence, including language features that allow a canister's data to be preserved when you upgrade the code of the canister.
```

> For example, Motoko lets you declare certain variables as `stable`. The values of `stable` variables are automatically preserved across software upgrades.

Reviewer: ```suggestion
For example, Motoko lets you declare certain variables as `stable`. The values of `stable` variables are automatically preserved across canister upgrades.
```

Because `+value+` was declared `stable`, the current state, _n_, of the service is retained after the upgrade. Counting will continue from _n_, not restart
from `0`.

> Because the new interface is compatible with the previous one, existing clients referencing the canister smart contract will continue to work, but new clients will be able to exploit its upgraded functionality

Reviewer: ```suggestion
Because the new interface is compatible with the previous one, existing clients referencing the canister will continue to work, but new clients will be able to exploit its upgraded functionality
```

{proglang} provides many other developer productivity features, including subtyping, arbitrary precision arithmetic and garbage collection.

{proglang} is not, and is not intended to be, the only language for implementing canister smart contracts.

> If it doesn't suit your needs, we are also working on adding Rust support to the SDK.

Reviewer: ```suggestion
If it doesn't suit your needs, there is a canister smart contract development kit for the Rust programming language.
```
